packages/ra-input-rich-text/src/index.spec.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import debounce from 'lodash/debounce';
+import {render, fireEvent, waitForElement, cleanup, getByTestId} from 'react-testing-library'
+
+import { RichTextInput } from './index';
+
+let container;
+
+jest.mock('lodash/debounce');
+describe('RichTextInput', () => {
+    beforeEach(() => {
+      container = document.createElement('div');
+      document.body.appendChild(container);
+      document.getSelection = () => {
+          return {
+            removeAllRanges: () => {},
+            getRangeAt: function() {}, 
+          };
+      };
+      jest.mock('lodash.debounce', (fn) => fn);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(container);
+        container = null;
+    });
+
+    it('should call text-change event only once when editing', async () => {
+        const mockFn = jest.fn();
+        debounce.mockImplementation(fn => fn);
+        const { getByTestId, rerender } = render(
+        <RichTextInput
+            input={{
+              value: '<p>test</p>',
+                onChange: mockFn
+            }}
+            meta={{error: null}} />);
+        const quillNode = await waitForElement(() => {
+            return getByTestId('quill')
+        });
+        quillNode.__quill.setText('test1');
```

Comment:
If you're using react-testing-library, this is not the way to go. You should mimick the actions of a real user, not use internal APIs.

Reply:
actually, i did tried the normal user way, but quilljs is using `contenteditable` and not input, thus not sure how to simulate a change event. Do let me know if you have any ideas.

Reply:
You can focus() on the element, then send key events. See https://stackoverflow.com/questions/6139107/programmatically-select-text-in-a-contenteditable-html-element

Reply:
hmm, just realised fireEvent actually works, it was due to the event loop. Had made the changes